smartsim/_core/_cli/build.py

```diff
@@ -285,124 +225,181 @@ def build_redis_ai(
         logger.info("ML Backends and RedisAI build complete!")
 
 
-def infer_torch_device() -> str:
-    backend_torch_path = f"{CONFIG.lib_path}/backends/redisai_torch"
-    device = "cpu"
-    if Path(f"{backend_torch_path}/lib/libtorch_cuda.so").is_file():
-        device = "gpu"
-    return device
+def check_py_torch_version(versions: Versioner, device: _TDeviceStr = "cpu") -> None:
+    """Check Python environment for TensorFlow installation"""
+
+    if BuildEnv.is_macos():
+        device_suffix = ""
```

Comment:
For paranoia, throw a warning (at least) here if someone requests gpu on a Mac. Or maybe do it in the extras vehcking stage.

Reply:
good call!